Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLAutoProcess.py

```diff
@@ -314,168 +390,228 @@ def PyExec(self):
-        self.progress.report('Processing container transmission')
-        if process_container_transmission:
-            SANSILLReduction(Run=self.ctransmission,
-                             ProcessAs='Transmission',
-                             OutputWorkspace=container_transmission_name,
-                             AbsorberInputWorkspace=transmission_absorber_name,
-                             BeamInputWorkspace=transmission_beam_name,
-                             NormaliseBy=self.normalise,
-                             BeamRadius=self.radius)
-
-        [process_sample_transmission, sample_transmission_name] = needs_processing(self.stransmission, 'Transmission')
-        self.progress.report('Processing sample transmission')
-        if process_sample_transmission:
-            SANSILLReduction(Run=self.stransmission,
-                             ProcessAs='Transmission',
-                             OutputWorkspace=sample_transmission_name,
-                             AbsorberInputWorkspace=transmission_absorber_name,
-                             BeamInputWorkspace=transmission_beam_name,
-                             NormaliseBy=self.normalise,
-                             BeamRadius=self.radius)
-
-        absorber = self.absorber[i] if len(self.absorber) == self.dimensionality else self.absorber[0]
-        [process_absorber, absorber_name] = needs_processing(absorber, 'Absorber')
-        self.progress.report('Processing absorber')
-        if process_absorber:
-            SANSILLReduction(Run=absorber,
-                             ProcessAs='Absorber',
-                             NormaliseBy=self.normalise,
-                             OutputWorkspace=absorber_name)
-
-        beam = self.beam[i] if len(self.beam) == self.dimensionality else self.beam[0]
-        [process_beam, beam_name] = needs_processing(beam, 'Beam')
-        flux_name = beam_name + '_Flux' if not self.flux[0] else ''
-        self.progress.report('Processing beam')
-        if process_beam:
-            SANSILLReduction(Run=beam,
-                             ProcessAs='Beam',
-                             OutputWorkspace=beam_name,
-                             NormaliseBy=self.normalise,
-                             BeamRadius=self.radius,
-                             AbsorberInputWorkspace=absorber_name,
-                             FluxOutputWorkspace=flux_name)
-
-        if self.flux[0]:
-            flux = self.flux[i] if len(self.flux) == self.dimensionality else self.flux[0]
-            [process_flux, flux_name] = needs_processing(flux, 'Flux')
-            self.progress.report('Processing flux')
-            if process_flux:
-                SANSILLReduction(Run=flux,
+    def processTransmissions(self):
+        with AutoProcessContext(self.atransmission, 'Absorber') as \
+                [process_transmission_absorber, transmission_absorber_name]:
+            self.progress.report('Processing transmission absorber')
```

Comment:
The split looks fine, the system test still passes. I wonder if we could somehow push the progress reports inside the ifs ? That way it will be more realistic and the number of progress reports will be actually proportional to what the algorithm would do. This is not critical and can be done in a separate PR though.